### ISSUE TITLE
No need to log the xlog standby with info running_xacts

### ISSUE THREAD

**Issue by paul-guo-** [greenplum-db/gpdb#9176]
Wu Hao found that there is such kind of xlog, 

`rmgr: Standby     len (rec/tot):     24/    50, tx:          0, lsn: 1/C0000028, prev 1/BCF8BE30, desc: RUNNING_XACTS nextXid 26262 latestCompletedXid 26261 oldestRunningXid 26262`

We support wal level WAL_LEVEL_REPLICA but do not support hot standby so this xlog does not seem to be useful. Not sure how frequently this xlog could be generated but we should disable it if can to save the xlog volume.

We probably need to check all callers of LogStandbySnapshot() and disable them one by one if needed.

Note gpdb master (actually pg9.5/9.6) combines WAL_LEVEL_HOT_STANDBY and WAL_LEVEL_ARCHIVE into WAL_LEVEL_REPLICA so it is possible that this issue does not exist in gpdb6 (due to some related condition check).

**Comment by ashwinstar** [comment]
It's unfortunate `WAL_LEVEL_REPLICA` doesn't allow to differentiate between wal to be generated for hot standby or not. If replica is present generate specific records seems to be the assumption of upstream code. Replica ignores the records during redo if its not hot. But primary side always generates the records.

Also, note `LogStandbySnapshot()` is not the only function, all records with  `RM_STANDBY_ID` RMGR type don't need to be emitted by GPDB till hot-standby is disabled. Plus, there are records under existing RMGR types also which are required only if hot standby is present.

Mostly looking into code under `XLogStandbyInfoActive()` gives picture of what's needed for replica with hot standby, maximum code under it seems to be for hot standby only. But given postgres documentation of `WAL_LEVEL_REPLICA` that can't be assumed and hence makes the situation dirtier if wish to disable only hot standby related ones. Though definitely seems quite a few extensive logging gets enabled due to hot standby like invalidation messages and exclusive locks and such. So, seems would be good to avoid the generation of same, as total wastage of resources in GPDB.

**Comment by ashwinstar** [comment]
Just for reference read this upstream thread https://www.postgresql.org/message-id/flat/55E50FD0.20607%40gmx.net and more forks from the same. Seems getting rid of archive vs hot_standby was for user experience perspective. There are mentions of doesn't matter between these two modes, but I don't see any numbers posted for amount of wal generation differences between these two modes. Hence, doesn't help us to easily know the impact of generating these wal records. In-case someone finds the thread which published numbers between these two settings, can help cut down our work to determine.

**Comment by gaos1** [comment]
in additional to xlog volumn, maybe it is also helpful to improve TPS for TP workload like TPC-B.  but we might also need to support hot standby. will this optimization cause code complex or diverge from upstream?

**Comment by paul-guo-** [comment]
> in additional to xlog volumn, maybe it is also helpful to improve TPS for TP workload like TPC-B. but we might also need to support hot standby. will this optimization cause code complex or diverge from upstream?

I do not think so. Note this issue might not exist on gpdb6 (not confirmed though) so if we do we might want to do after master merge is finished.

**Comment by soumyadeep2007** [comment]
b555ed8102672cfedb06559952b8341756386d69 removed the hot_standby wal_level and it raises no performance overhead concerns whatsoever.

Furthermore, it doesn't do that much harm to keep logging `XLOG_RUNNING_XACTS` WAL records, even if HS is disabled.

1. `XLOG_RUNNING_XACTS` is only logged once per checkpoint.
2. `XLOG_RUNNING_XACTS` is logged by the bgwriter at `LOG_SNAPSHOT_INTERVAL_MS (15s)` intervals.

However, there are these other 2 messages that have higher overhead:
1. `XLOG_INVALIDATIONS` logged for every commit involving a sinval operation.
2. `XLOG_STANDBY_LOCK` every time an `AccessExclusiveLock` is taken and also it is logged together with `XLOG_RUNNING_XACTS`.
